mmpose/models/heads/cid_head.py

```diff
@@ -0,0 +1,337 @@
+            maxm = F.max_pool2d(heatmap, 7, 1, 3)
+        elif map_size > 200:
+            maxm = F.max_pool2d(heatmap, 5, 1, 2)
+        else:
+            maxm = F.max_pool2d(heatmap, 3, 1, 1)
+        return maxm
+
+    def init_weights(self):
+        """Initialize model weights."""
+        for m in self.modules():
+            if isinstance(m, nn.Conv2d):
+                nn.init.normal_(m.weight, std=0.001)
+                for name, _ in m.named_parameters():
+                    if name in ['bias']:
+                        nn.init.constant_(m.bias, 0)
+            elif isinstance(m, nn.Linear):
+                nn.init.normal_(m.weight, std=0.001)
+                for name, _ in m.named_parameters():
+                    if name in ['bias']:
+                        nn.init.constant_(m.bias, 0)
+        # focal loss init
+        bias_value = -math.log((1 - self.prior_prob) / self.prior_prob)
+        torch.nn.init.constant_(self.keypoint_center_conv.bias, bias_value)
+        torch.nn.init.constant_(self.heatmap_conv.bias, bias_value)
+
+class ChannelAtten(nn.Module):
+    def __init__(self, in_channels, out_channels):
+        super(ChannelAtten, self).__init__()
+        self.atn = nn.Linear(in_channels, out_channels)
+
+    def forward(self, global_features, instance_params):
+        B, C, H, W = global_features.size()
+        instance_params = self.atn(instance_params).reshape(B, C, 1, 1)
+        return global_features * instance_params.expand_as(global_features)
+
+class SpatialAtten(nn.Module):
+    def __init__(self, in_channels, out_channels):
+        super(SpatialAtten, self).__init__()
+        self.atn = nn.Linear(in_channels, out_channels)
+        self.feat_stride = 4
+        conv_in = 3
+        self.conv = nn.Conv2d(conv_in, 1, 5, 1, 2)
+
+    def forward(self, global_features, instance_params, instance_inds):
+        B, C, H, W = global_features.size()
+        instance_params = self.atn(instance_params).reshape(B, C, 1, 1)
+        feats = global_features * instance_params.expand_as(global_features)
+        fsum = torch.sum(feats, dim=1, keepdim=True)
+        input_feats = fsum
+        locations = compute_locations(global_features.size(2), global_features.size(3), stride=1, device=global_features.device)
+        n_inst = instance_inds.size(0)
+        H, W = global_features.size()[2:]
+        instance_locations = torch.flip(instance_inds, [1])
+        instance_locations = instance_locations
+        relative_coords = instance_locations.reshape(-1, 1, 2) - locations.reshape(1, -1, 2)
+        relative_coords = relative_coords.permute(0, 2, 1).float()
+        relative_coords = (relative_coords / 32).to(dtype=global_features.dtype)
+        relative_coords = relative_coords.reshape(n_inst, 2, H, W)
+        input_feats = torch.cat((input_feats, relative_coords), dim=1)
+        mask = self.conv(input_feats).sigmoid()
+        return global_features * mask
+
+class FocalLoss(nn.Module):
```

Comment:
How about moving these Loss to `mmpose/models/losses`?
For example, https://github.com/open-mmlab/mmpose/blob/master/mmpose/models/losses/mse_loss.py

Reply:
I move the FocalLoss (renamed as FocalHeatmapLoss) to mmpose/models/losses module, because this loss can also be used by other projects. I did not move the contrastive loss because I think it may be rarely used by other projects and moving it will cause redundancy.